rust/private/repository_utils.bzl

```diff
@@ -463,12 +463,20 @@ def produce_tool_suburl(tool_name, target_triple, version, iso_date = None):
         target_triple: The rust-style target triple of the tool
         version: The version of the tool among "nightly", "beta', or an exact version.
         iso_date: The date of the tool (or None, if the version is a specific version).
-    """
 
+    Returns:
+        str: The fully qualified url path for the specified tool.
+    """
+    url = ""
     if iso_date:
-        return "{}/{}-{}-{}".format(iso_date, tool_name, version, target_triple)
-    else:
-        return "{}-{}-{}".format(tool_name, version, target_triple)
+        url += "{}/".format(iso_date)
+    if tool_name:
+        url += tool_name
```

Comment:
Tool is not an optional argument, let's`fail()` with an error message when None is passed. Same for version.

Reply:
Done.